src/sentry/integrations/source_code_management/repository.py

```diff
@@ -132,12 +132,13 @@ def check_file(self, repo: Repository, filepath: str, branch: str | None = None)
             except IdentityNotValid:
                 return None
             except ApiError as e:
-                if e.code != 404:
+                if e.code in (404, 400):
+                    lifecycle.record_halt(extra={"status_code": e.code})
+                    return None
```

Comment:
One word of caution for this, we have a scenario in gitlab where I think we're failing to provide the correct branch information but I don't remember the full context on this (Here's the PR we added logging to check for this though: https://github.com/getsentry/sentry/pull/81354). Let's merge this and monitor these lifecycle logs after.

Reply:
hmm perhaps i should be logging the exception itself for more information